### ISSUE TITLE
Cannot set FK to null when using @ManyToOne() + @Column() after promise resolution

### ISSUE THREAD

**Issue by parisholley** [typeorm/typeorm#7144]
### Expected Behavior

As described here:

https://github.com/typeorm/typeorm/issues/1795

The library supports being able to initially set the value using the id property instead of doing `relation = Promise.resolve(entity)`. I would expect to also be able to do the follow to "unset" the value:

```
/// hydrate the object
await entity.relation;

entity.relationId = null;
await em.save(entity)'
```

### Actual Behavior

The problem is, there is no way to unset that value (nor I suspect to update it) if the promise property has been touched. The way the diff detection works:

https://github.com/typeorm/typeorm/blob/f27622daf9784d06496975bbbcacdb0938fd2c82/src/persistence/SubjectChangedColumnsComputer.ts#L56-L74

The default behavior of `getEntityValue` is to always pull the promise'd value if it is available and the column you are touching is related to another entity. Even though `entity[this.propertyName]` is set to null, it gets ignored.

I've tried a few things config wise and tweaking the code but i'm not familiar enough with the library to know what the right answer is.

`{persistence:true}` is not read by the diffing code and adding checks for it causes other unintended consequences. The only way to work around this right now, is to switch to property setters in my model that always nulls out the promise if it is around, but that is not ideal.

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [x] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by nebkat** [comment]
Is it the `(value !== null && value !== undefined)` that is causing it to be ignored? If so we should surely treat `undefined` as an absence of value and ignore, while `null` means the value is set to null, which is their purpose in JavaScript in general.

**Comment by parisholley** [comment]
no, because the @Column() property has a relation due to the @ManyToOne() mapping on the same column name, `this.relationMetadata && this.referencedColumn` is true, and it will always  `this.referencedColumn.getEntityValue(relatedEntity)`, which is the promise value, and not entity[this.propertyName], which was explicitly set to `NULL` to remove

**Comment by parisholley** [comment]
so to clarify, it isn't that it is being skipped and hitting the falsy check, it is comparing the wrong value to the database entity, and things the user hasn't null'd it out

**Comment by isaachy** [comment]
Hello, I am starting a long-term project and I am thinking whether to choose this ORM or not, due to these problems, do you recommend its use? Or maybe I should use Prisma

**Comment by ngouy** [comment]
Same issue if you use eager load
aka if you eager load your one to many relation, then set to a differentId, `.save` will completely ignore it and won't save the new relationship id

**Comment by r3ps4J** [comment]
I just ran into this. I fixed it by manually retrieving the relation in my case with `customerId` in the method where I update the entity. This way I can still access the old customer for logging purposes and update the relation by setting `customerId` to the id of the new customer.

**Comment by kendallroth** [comment]
Just ran into this again, and thought I'd bring it back up 🤷. If manually specifying either a `null` value for a relational ID field, or changing it to another value, I would expect the relationship to be updated (not ignored), rather than having to fetch/set the relational object instead.
